### ISSUE TITLE
Incorrect symbol highlight in anonymous function with `case`

### ISSUE THREAD

**Issue by filipwiech** [scalameta/metals#5863]
### Describe the bug

It seems that in some cases (pun not intended) Metals is highlighting different symbols which happen to share the same name. I observed it when defining anonymous partial function as a body of a parameter which takes a normal function. It doesn't seem to happen if the parameter expects a partial function explicitly. Example (@@ marks the cursor position):
```scala
// works correctly, collect takes PartialFunction
List("test").collect {
    case string@@Name: String if stringName.startsWith("a") => stringName + "a"
    case stringName: String if stringName.startsWith("b") => stringName + "b"
}

// doesn't work as expected, map and foldLeft take normal function
List("test").map {
      case string@@Name: String if stringName.startsWith("a") => stringName + "a"
      case stringName: String if stringName.startsWith("b") => stringName + "b"
      case stringName: String => stringName + "c"
}

List("test").foldLeft("abc"){
    case (acc: String, na@@me: String) if name.startsWith("a") => acc + "a" + name
    case (acc: String, name: String) if na@@me.startsWith("b") => acc + "b" + name
    case (acc: String, name: String) => a@@cc + "c" + name
}
```
Tested with Scala 2.13.

### Expected behavior

Only the exact same symbols are highlighted, even if they share the name with others in the scope of the overall function.

### Operating system

Linux

### Editor/Extension

VS Code

### Version of Metals

1.1.0+97-57eb284c-SNAPSHOT

### Extra context or search terms

highlight, highlighting, anonymous function, case, partial function

**Comment by tgodzik** [comment]
Thanks for reporting! We noticed that myself, we'll take a look into it!

**Comment by filipwiech** [comment]
I can confirm that the issue is fixed after the #5876. :+1: Thank you @jkciesluk so much for a quick fix! :slightly_smiling_face:
